Sources/Citadel/TTY/Client/TTY.swift

```diff
@@ -222,20 +227,30 @@ extension SSHClient {
     /// Executes a command on the remote server. This will return the output stream of the command. If the command fails, the error will be thrown.
     /// - Parameters:
     /// - command: The command to execute.
-    public func executeCommandStream(_ command: String) async throws -> AsyncThrowingStream<ExecCommandOutput, Error> {
+    /// - inShell:  Whether to request the remote server to start a shell before executing the command.
+    public func executeCommandStream(_ command: String, inShell: Bool = false) async throws -> AsyncThrowingStream<ExecCommandOutput, Error> {
         var streamContinuation: AsyncThrowingStream<ExecCommandOutput, Error>.Continuation!
         let stream = AsyncThrowingStream<ExecCommandOutput, Error> { continuation in
             streamContinuation = continuation
         }
         
-        let handler = ExecCommandHandler(logger: logger) { output in
+        var hasReceivedChannelSuccess = false
+        
+        let handler = ExecCommandHandler(logger: logger) { channel, output in
             switch output {
             case .stdout(let stdout):
                 streamContinuation.yield(.stdout(stdout))
             case .stderr(let stderr):
                 streamContinuation.yield(.stderr(stderr))
             case .eof(let error):
                 streamContinuation.finish(throwing: error)
+            case .channelSuccess:
+                if inShell, !hasReceivedChannelSuccess {
+                    let commandData = SSHChannelData(type: .channel,
+                                                     data: .byteBuffer(ByteBuffer(string: command + ";exit\n")))
+                    channel.writeAndFlush(commandData, promise: nil)
+                    hasReceivedChannelSuccess = true
```

Comment:
My one nit is that `hasReceivedChannelSuccess` should be always set, otherwise if we access that variable in a later feature down the line it will cause confusion

Reply:
I’m wondering why “exit” is always called after user’s command?

Reply:
@xuxulll because this is a helper that only executes a _single_ command.